app/views/layouts/application.html.erb

```diff
@@ -9,11 +9,44 @@
   </head>
 
   <body>
+    <% if session[:user_id] != nil %>
+      <% user = User.find(session[:user_id]) %>
+    <% end %>
     <nav class = "topnav">
-      <%= link_to "All Merchants", "/merchants"%>
-      <%= link_to "All Items", "/items"%>
-      <%= link_to "Cart: #{cart.total_items}", "/cart" %>
-      <%= link_to "Register", "/register" %>
+      <% if user == nil %>
+        <%= link_to "Home", "/" %>
+        <%= link_to "All Merchants", "/merchants"%>
+        <%= link_to "All Items", "/items"%>
+        <%= link_to "Cart: #{cart.total_items}", "/cart" %>
+        <%= link_to "Register", "/register" %>
+        <%= link_to "Log in", "/login" %>
+      <% elsif user.role == "default" %>
+        <%= link_to "Home", "/" %>
+        <%= link_to "All Merchants", "/merchants"%>
+        <%= link_to "All Items", "/items"%>
+        <%= link_to "Cart: #{cart.total_items}", "/cart" %>
+        <%= link_to "Log out", "/logout" %>
+        <%= link_to "Profile", "/profile" %>
+        <%= "Logged in as #{user.name}" %>
+      <% elsif user.role == "admin" %>
+        <%= link_to "Home", "/" %>
+        <%= link_to "All Merchants", "/merchants"%>
```

Comment:
Maybe not right this second, but eventually I would love for these paths to be namespaced (am I using that properly?) so that we can have a cleaner page.